gui/quantum.lua

```diff
@@ -11,262 +11,218 @@ local quickfort = reqscript('quickfort')
-end
-
-function QuantumUI:onRenderBody()
-    if not self.feeder then return end
-
-    self:render_feeder_overlay()
-    self:render_destination_overlay()
-end
-
-function QuantumUI:onInput(keys)
-    if self:inputToSubviews(keys) then return true end
-
-    self:propagateMoveKeys(keys)
-
-    local pos = nil
-    if keys._MOUSE_L_DOWN then
-        pos = dfhack.gui.getMousePos()
-        if pos then
-            guidm.setCursorPos(pos)
-        end
-    elseif keys.SELECT then
-        pos = guidm.getCursorPos()
-    end
-
-    if pos then
-        if not self.feeder then
-            self:select_stockpile(pos)
-        else
-            local qsp_pos = self:get_qsp_pos(pos)
-            if not is_valid_pos(pos, qsp_pos) then
-                return
-            end
-
-            self:dismiss()
-            self:commit(pos, qsp_pos)
-        end
-    end
-end
-
-local function get_feeder_pos(feeder_tiles)
-    for z,rows in pairs(feeder_tiles) do
-        for y,row in pairs(rows) do
-            for x in pairs(row) do
-                return xyz2pos(x, y, z)
-            end
-        end
-    end
-end
-
-local function get_moves(move, move_back, start_pos, end_pos,
-                         move_to_greater_token, move_to_less_token)
-    if start_pos == end_pos then
-        return move, move_back
-    end
-    local diff = math.abs(start_pos - end_pos)
-    local move_to_greater_pattern = ('{%s %%d}'):format(move_to_greater_token)
-    local move_to_greater = move_to_greater_pattern:format(diff)
-    local move_to_less_pattern = ('{%s %%d}'):format(move_to_less_token)
-    local move_to_less = move_to_less_pattern:format(diff)
-    if start_pos < end_pos then
-        return move..move_to_greater, move_back..move_to_less
-    end
-    return move..move_to_less, move_back..move_to_greater
-end
+    quickfort.apply_blueprint{
+        mode='build',
+        pos=dfhack.gui.getMousePos(),
+        data={
+            [0]={
+                [0]={[0]=placed_trackstop}
+            }
+        }
+    }
 
-local function get_quantumstop_data(dump_pos, feeder_pos, name)
-    local move, move_back = get_moves('', '', dump_pos.z, feeder_pos.z, '<','>')
-    move, move_back = get_moves(move, move_back, dump_pos.y, feeder_pos.y,
-                                'Down', 'Up')
-    move, move_back = get_moves(move, move_back, dump_pos.x, feeder_pos.x,
-                                'Right', 'Left')
-
-    local quantumstop_name_part, quantum_name_part = '', ''
-    if name ~= '' then
-        quantumstop_name_part = (' name="%s quantum"'):format(name)
-        quantum_name_part = ('{givename name="%s dumper"}'):format(name)
+    if
+        cart_count == 0
+    then
+        dfhack.run_command('workorder "{\"amount_left\":1,\"amount_total\":1,\"frequency\":\"OneTime\",\"id\":6,\"is_active\":false,\"is_validated\":true,\"item_subtype\":\"ITEM_TOOL_MINECART\",\"job\":\"MakeTool\",\"material_category\":[\"wood\"]}"')
```

Comment:
why are you ordering a minecart via `workorder` instead of the `order_minecart` function below?

Reply:
Basic answer was it was what I could get to work. The order Minecart function was one of the things I couldn't seem to get to work properly before, but now that the class stuff's been sorted out I'll see what I can do about it. I chose work order specifically because it wasn't reliant on any external blueprint or work order file and I could use a json string in the script itself. I can change it if you want me to.